sw/device/tests/sim_dv/sysrst_ctrl_in_irq_test.c

```diff
@@ -0,0 +1,234 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_pinmux.h"
+#include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/dif/dif_sysrst_ctrl.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/irq.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+static dif_sysrst_ctrl_t sysrst_ctrl;
+static dif_rv_plic_t plic;
+
+enum {
+  kTestPhaseTimeoutUsec = 20,
+  kPlicTarget = kTopEarlgreyPlicTargetIbex0,
+};
+
+static volatile dif_sysrst_ctrl_irq_t irq;
+static volatile top_earlgrey_plic_peripheral_t peripheral;
+
+typedef enum test_phases {
+  phaseInitial = 0,
+  phaseBeforeInputChange = 1,
+  phaseAfterInputChange = 2,
+  phaseBeforeComboChange0 = 3,
+  phaseAfterComboChange0 = 4,
+  phaseBeforeComboChange1 = 5,
+  phaseAfterComboChange1 = 6,
+  phaseDone = 7,
+} test_phases_t;
+
+// Test phase written by testbench.
+static volatile const uint8_t kTestPhase = 0;
+test_phases_t phase = phaseInitial;
+
+enum {
+  kOutputNumPads = 0x8,
+  kOutputNunMioPads = 0x6,
+};
+
+static const dif_pinmux_index_t kPeripheralInputs[] = {
+    kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonKey0In,
+    kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonKey1In,
+    kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonKey2In,
+    kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonPwrbIn,
+    kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonAcPresent,
+    kTopEarlgreyPinmuxPeripheralInSysrstCtrlAonLidOpen,
+};
+
+static const dif_pinmux_index_t kInputPads[] = {
+    kTopEarlgreyPinmuxInselIob3, kTopEarlgreyPinmuxInselIob6,
+    kTopEarlgreyPinmuxInselIob8, kTopEarlgreyPinmuxInselIor13,
+    kTopEarlgreyPinmuxInselIoc7, kTopEarlgreyPinmuxInselIoc9,
+};
+
+static const dif_sysrst_ctrl_pin_t kSysrstCtrlInputs[] = {
+    kDifSysrstCtrlPinKey0In,           kDifSysrstCtrlPinKey1In,
+    kDifSysrstCtrlPinKey2In,           kDifSysrstCtrlPinPowerButtonIn,
+    kDifSysrstCtrlPinAcPowerPresentIn, kDifSysrstCtrlPinLidOpenIn,
+    kDifSysrstCtrlPinEcResetInOut,     kDifSysrstCtrlPinFlashWriteProtectInOut,
+};
+
+void test_phase_sync() {
+  test_status_set(kTestStatusInTest);
+  test_status_set(kTestStatusInWfi);
+}
+
+/**
+ * Sync with DV side, wait for interrupt for input change, check the interrupt
+ * cause and clear interrupt cause
+ */
+void sysrst_ctrl_input_change_detect(
```

Comment:
I didn't looked deeply enought, but can this function be merged with `sysrst_ctrl_input_change_combo_detect`?

Reply:
I think it is simpler to read and understand if we seperate input change and key combo detect functions.